iree/turbine/kernel/wave/utils.py

```diff
@@ -670,6 +696,54 @@ def compile_to_vmfb(
 RUNTIME_CACHE: dict[str, tuple[rt.SystemContext, rt.VmFunction]] = {}
 
 
+def invoke_with_wave_runtime(
+    kernel_inputs: list[torch.Tensor],
+    kernel_outputs: list[torch.Tensor],
+    kernel_dynamic_dims: list[int],
+    kernel_hash: str,
+    kernel_info: KernelLaunchInfo,
+):
+    """
+    Invokes the kernel with the wave runtime.
+    """
+    import wave_runtime
+    from .cache import WAVE_RUNTIME_DIR, CACHE_BASE_DIR
+
+    # Get the path to the binary.
+    if kernel_hash:
+        binary = str(CACHE_BASE_DIR / kernel_hash / kernel_hash) + ".hsaco"
+    else:
+        bin_dir = WAVE_RUNTIME_DIR
+        binary = glob.glob(str(bin_dir / "*.hsaco"))[0]
```

Comment:
similar Q to previous one, can you explain the mechanism behind selecting the first found file in WAVE_RUNTIME_DIR/*.hsaco?

Also NIT: what about `binary = glob.glob(str(WAVE_RUNTIME_DIR / "*.hsaco"))[0]`?

Reply:
By removing the last compile file we make sure that there will always be only one kernel inside the wave runtime directory inside the cache directory. That is why it's okay for us to just use the first file that we find. Sure I can modify that.